### PR TITLE
Initial commit of A3 blueprint and documentation

### PR DESCRIPTION
**Pull request by tpdownes**
This commit adds the first Toolkit solution for provisioning Slurm clusters with A3 compute nodes each with 8 NVIDIA H100 GPUs. The associated provisioning guide documents for the user how to create base networking infrastructure, build a custom image, and provision the A3 cluster itself.

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/machine-learning/README.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+benchmark. It assumes the availability of a GPU/NIC topology file at
+`/opt/tcpdirect_benchmark/gpu_rxq_configuration.textproto`. This file is built
+into the DLVM images used by this solution, but may need to be provided if
+using an alternative image.
+
+### Clone the HPC Toolkit repository containing the NCCL benchmark
+
+```shell
+git clone https://github.com/GoogleCloudPlatform/hpc-toolkit
+cd hpc-toolkit/examples/machine-learning/nccl-tests
```
Comment (harshthakkar01) [inline examples/machine-learning/README.md:300]: Do we have this in repo or we are planning to add in following PRs ?

**Comment by harshthakkar01** [comment]
PR-validation failure:
```
Step #1 - "make-tests": Error: deployment variable "server_ip_homefs" was not set
Step #1 - "make-tests": 51:   server_ip_homefs:
Step #1 - "make-tests":       ^
Step #1 - "make-tests": *** ERROR: error creating deployment with ghpc for ml-slurm-h100-2-cluster.yaml
Step #1 - "make-tests": *** ERROR: a test failed. Exiting with status 1.
Step #1 - "make-tests": make: *** [Makefile:153: validate_configs] Error 1
```

File: examples/machine-learning/README.md (hunk 1 above)
Comment (tpdownes) [reply examples/machine-learning/README.md:300]: These are contained in #2361

File: examples/machine-learning/README.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
+# Objective
```
Comment (cboneti) [inline examples/machine-learning/README.md:1]: One thought now, I wonder if we should add A3 somewhere here
today, we have: examples/machine-learning/ml-slurm-h100-2-cluster.yaml
I am wondering how would have to change for different machine types.

**Review by cboneti — CHANGES_REQUESTED** [review]

**Comment by tpdownes** [comment]
> PR-validation failure:
> 
> ```
> Step #1 - "make-tests": Error: deployment variable "server_ip_homefs" was not set
> Step #1 - "make-tests": 51:   server_ip_homefs:
> Step #1 - "make-tests":       ^
> Step #1 - "make-tests": *** ERROR: error creating deployment with ghpc for ml-slurm-h100-2-cluster.yaml
> Step #1 - "make-tests": *** ERROR: a test failed. Exiting with status 1.
> Step #1 - "make-tests": make: *** [Makefile:153: validate_configs] Error 1
> ```

Resolved.

**Review by harshthakkar01 — APPROVED** [review]

**Review by cboneti — APPROVED** [review]

**Pull request merged**